var/spack/repos/builtin/packages/xerces-c/package.py

```diff
@@ -37,5 +37,11 @@ class XercesC(AutotoolsPackage):
 
     version('3.1.4', 'd04ae9d8b2dee2157c6db95fa908abfd')
 
+    depends_on('libiconv')
+
+    def setup_environment(self, spack_env, run_env):
+        spack_env.append_flags('LDFLAGS', '-L{0} -liconv'.format(
+            self.spec['libiconv'].prefix.lib))
```

Comment:
You could try using `self.spec['libiconv'].libs.ld_flags` here, which will return the full path and all link flags.

Reply:
It works after renaming `libiconv` to `iconv` → `_libs_default_handler` adds another `lib` prefix (i.e. `liblibiconv` instead of `libiconv`):

```Python
    def setup_environment(self, spack_env, run_env):
        self.spec['libiconv'].name = 'iconv'
        spack_env.append_flags('LDFLAGS', self.spec['iconv'].libs.ld_flags)
        self.spec['iconv'].name = 'libiconv'
```

Reply:
I would just override the `libs` attribute of libiconv. You can check BLAS/LAPACK/MPI packages for how to do this. But a "no double lib" heuristic in the default handler might not be a bad idea.

Reply:
I implemented your suggestion in #7651. You can rebase once it is merged.

Reply:
Okay, #7651 has been merged. Can you rebase and try again?

Reply:
Yes, it seems to work → I'll wait for the build to complete and force-push the proposed change.

Reply:
done